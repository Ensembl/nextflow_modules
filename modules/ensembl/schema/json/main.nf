// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process SCHEMA_JSON {
    tag "$meta.id"
    label 'process_low'

    container "ensemblorg/ensembl-genomio:v1.6.1"

    input:
        tuple val(meta), path(json_file)

    output:
        tuple val(meta), path(json_file), emit: verified_json
        path "versions.yml", emit: versions

    when:
        task.ext.when == null || task.ext.when

    script:
        schema_name = json_file.simpleName
        """
        schemas_json_validate --json_file !{json_file} --json_schema !{schema_name}

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        schemas_json_validate --version >> versions.yml
        """

    stub:
        schema_name = json_file.simpleName
        """
        echo "No change, don't create schema error log"

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        schemas_json_validate --version >> versions.yml
        """
}
