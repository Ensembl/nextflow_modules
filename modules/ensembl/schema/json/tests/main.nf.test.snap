{
    "Schema check for the JSON file stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "meta.json:md5,62eec34a06788123009463246888d349"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a64c78004731777f7f8143342217afcf"
                ],
                "verified_json": [
                    [
                        {
                            "id": "test"
                        },
                        "meta.json:md5,62eec34a06788123009463246888d349"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a64c78004731777f7f8143342217afcf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-22T14:17:48.70924"
    }
}